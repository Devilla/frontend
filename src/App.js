import React from 'react';
import { browserHistory } from 'react-router';
import { AppContainer } from 'react-hot-loader';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import rootSaga from './sagas';
import reducer from './ducks';
import thunk from 'redux-thunk';
import { syncHistoryWithStore } from 'react-router-redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import createSagaMiddleware from 'redux-saga';
import { StripeProvider } from 'react-stripe-elements';

const sagaMiddleware = createSagaMiddleware();

export const store = createStore(reducer, composeWithDevTools(
  applyMiddleware(sagaMiddleware, thunk)
));

const routerHistory = syncHistoryWithStore(browserHistory, store, {selectLocationState: state => state.get('router')});


sagaMiddleware.run(rootSaga);

const App = ({Component}) =>
  <AppContainer errorReporter={Error}>
    <Provider store={store}>
      <StripeProvider apiKey={process.env.NODE_ENV === 'production' ? process.env.REACT_APP_STRIPE_KEY : process.env.REACT_APP_DEVELOPMENT_STRIPE_KEY}>
        <Component routerHistory={routerHistory} />
      </StripeProvider>
    </Provider>
  </AppContainer>;

export default App;
