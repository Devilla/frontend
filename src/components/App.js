import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import  '../assets/stylesheets/slick.css'
import  '../assets/stylesheets/slick-theme.css'
import  '../assets/stylesheets/style.css'

import Header from './_common/Header';
import Footer from './_common/Footer';
import { Spinner } from './index.js';

class App extends Component {
  render(){
    return (
        <div className="app-container">
          <Header />

          <Spinner loading={this.props.loading} />
          <div className="content">
            {this.props.children}
          </div>
          <Footer />
        </div>
    );
  }
};

App.propTypes = {
  children: PropTypes.node
};

const mapStateToProps = state => ({
  loading: state.getIn(['loading', 'state'])
});

export default connect(mapStateToProps)(App);
