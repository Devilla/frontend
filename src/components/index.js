export { default as WebsiteHeader } from "./WebsiteHeader/WebsiteHeader";
export { default as WebsiteFooter } from "./WebsiteFooter/WebsiteFooter";
export { default as WebsiteHome } from "./WebsiteHome/WebsiteHome";
export { default as WebsiteHowItWorks } from "./WebsiteHowItWorks/WebsiteHowItWorks";
export { default as WebsiteIntegrations } from "./WebsiteIntegrations/WebsiteIntegrations";
export { default as WebsitePricing } from "./WebsitePricing/WebsitePricing";
export { default as WebsiteAbout } from "./WebsiteAbout/WebsiteAbout";
export { default as WebsiteSignIn } from "./WebsiteSignIn/WebsiteSignIn";
export { default as WebsiteSignUp } from "./WebsiteSignUp/WebsiteSignUp";
export { default as WebsiteContact } from "./WebsiteContact/WebsiteContact";
export { default as WebsiteTerms } from "./WebsiteTerms/WebsiteTerms";
export { default as WebsitePolicy } from "./WebsitePolicy/WebsitePolicy";
export { default as WebsiteDemoPage}  from "./WebsiteDemoPage/WebsiteDemoPage";
export { default as WebsiteCustStory}  from "./WebsiteCustStory/WebsiteCustStory";
export { default as WebsiteFeature}  from "./WebsiteFeature/WebsiteFeature";
export { default as Affiliate}  from "./Affiliate/Affiliate";
export { default as AffiliateRegister}  from "./Affiliate/AffiliateRegister";

export { default as Header } from "./Header/Header";
export { default as ForgetPassword } from "./ForgetPassword";
export { default as PricePage } from "./PricePage";
export { default as Common } from "./Common/function";
export { default as ResetPassword } from "./ResetPassword";
export { default as Sidebar } from "./Sidebar/Sidebar";
export { default as StatsCard } from "./StatsCard/StatsCard";
export { default as Dashboard } from './Dashboard/Dashboard';
export { default as Notification } from './Notification';
export { default as Analytics } from './Analytics/analytics';
export { default as AnalyticsProfile } from './Analytics/AnalyticsProfile';
export { default as New } from './New/new';
export { default as LoginFlow } from './LoginFlow/LoginFlow';
export { default as Spinner } from './Spinner/Spinner';
export { default as Error } from './Error/Error';
export { default as ComingSoon } from './ComingSoon/ComingSoon';
export { default as Button } from './Button/Button';
export { default as Profile } from './Profile/Profile';
export { default as BillingDetails } from './Billing/BillingDetails';
export { default as UpgradeCard } from './UpgradeCard/UpgradeCard';
export { default as UpgradePlan } from './UpgradePlan/UpgradePlan';
export { default as Integration } from './Integration/Integration';
export { default as Template } from './Template';
export { default as VerificationPage } from './VerificationPage/VerificationPage';
