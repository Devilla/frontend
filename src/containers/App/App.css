@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
}
