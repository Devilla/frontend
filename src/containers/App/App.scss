@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
  display: block;
  margin: 0px;
}

.website-app {
  .alert-warning {
    background-color: #f2dede;
    border-color: #ebccd1;
    color: #a94442;
  }
  .content {
    padding-top: 100px;
  }
  .toast-style {
    padding: 17px 7px;
    background-color: rgba(0, 0, 0, 0.9) !important;
    color: #fff !important;
    text-align: center !important;
    height: 10px !important;
    border-radius: 5px !important;
    button {
      height: 1.5em;
      color: #fff;
    }
  }

  .transition-wrapper {
    position: relative;
    z-index: 1;
    .transition-item {
      position: fixed;
      top: 0;
      right: 0;
      left: 0;
    }
  }
  .signin-container,
  .authpage,
  .websitepolicy-container,
  .websitehowitworks-container,
  .websitehome-container,
  .WebsiteTerms-container,
  .websitepricing-container,
  .website-payment-container,
  .websiteintegrations-container,
  .demopage-container,
  .website-cust-story,
  .websitecontact-container,
  .websiteabout-container,
  .WebsiteFeature-container,
  .affiliateregister-container
   {
    overflow: hidden;
    box-sizing: border-box;
    padding: 20px;
    min-height: 100vh;
    transition: 500ms ease-out;

    &.transition-leave {
      opacity: 1;
    }

    &.transition-leave.transition-leave-active {
      opacity: 1;
      transform: translate3d(100%, 0, 0);
    }
  }
}
