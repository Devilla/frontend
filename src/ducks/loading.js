import { Map } from 'immutable';

const action = name => `zikher/modals/${name}`;
const LOAD = action('LOAD');
const LOADED = action('LOADED');

export const load = () => ({ type: LOAD });
export const loaded = () => ({ type: LOADED });

const initialState = Map({
  state: false,
});

const loading = (state = initialState, action) => {
  switch (action.type) {
    case LOAD:
      return state.set('state', true);

    case LOADED:
      return state.set('state', false);

    default:
      return state;
  }
};

export default loading;
