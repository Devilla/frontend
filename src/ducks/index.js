import { combineReducers } from 'redux-immutable';
import { routerReducer } from 'react-router-redux'

import profile from './profile';
import auth from './auth';
import plan from './plan';
import payment from './payment';
import notification from './notification';
import campaign from './campaign';
import rules from './rules';
import loading from './loading';
import configuration from './configuration';
import elastic from './elastic';
import pageurl from './pageurl';
import webhooks from './webhooks';
import subcampaign from './subcampaign';

const reducer = combineReducers({
  profile,
  auth,
  plan,
  payment,
  notification,
  campaign,
  rules,
  loading,
  configuration,
  elastic,
  pageurl,
  webhooks,
  subcampaign,
  router: routerReducer
});

export default reducer;
