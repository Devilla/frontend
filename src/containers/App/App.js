import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ToastContainer } from 'react-toastify';
import { browserHistory } from 'react-router';
import { connect } from 'react-redux';
import { Spinner } from '../../components/index.js';
import { WebsiteHeader, WebsiteFooter } from 'components';

import { checkTokenExists } from 'ducks/auth';

import './scss/stack-interface.scss';
import './scss/socicon.css';
import './scss/iconsmind.scss';
import './scss/bootstrap.scss';
import './scss/theme.scss';
import './scss/custom.scss';
import './scss/font-raleway.scss';
import './scss/flickity.scss';
import './scss/font-sourcesanspro.scss';
import './App.scss';

class App extends Component {
  componentWillMount() {
    this.checkLogin((err) => {
      if(!err) {
        browserHistory.push('/dashboard');
      }
    });
  }

  checkLogin(callback) {
    const cookie = localStorage.getItem('authToken');
    const authToken = cookie
      ? JSON.parse(cookie)
      : null;
    if (authToken) {
      this.props.checkTokenExists(authToken);
      callback()
    } else {
      callback("not logged in")
    }
  }

  render(){
    return (
      <div className="website-app">
        <div className="basic-gradient-light" data-smooth-scroll-offset="77">
           <WebsiteHeader />
           <Spinner loading={this.props.loading} />
           <div className="content">
             {this.props.children}
           </div>
           <WebsiteFooter />
        </div>
        <ToastContainer hideProgressBar={true} />
      </div>
    );
  }

}

const mapDispatchToProps = {
  checkTokenExists
};

export default connect(null, mapDispatchToProps)(App);
