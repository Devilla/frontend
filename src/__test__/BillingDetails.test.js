import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import {createStore} from 'redux'
import { load, loaded } from 'ducks/loading';
import { BillingDetails } from 'components';
import { store } from 'App';

// Snapshot for Home React Component
describe('>>> BillingDetails --- Snapshot',()=>{
    it('+++capturing Snapshot of BillingDetails', () => {
        const renderedValue =  renderer.create(<BillingDetails store={store} />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});
