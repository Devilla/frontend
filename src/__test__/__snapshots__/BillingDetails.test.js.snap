// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> BillingDetails --- Snapshot +++capturing Snapshot of BillingDetails 1`] = `
<div
  className="content fill billing-details ml-1"
>
  <div
    className="container-fluid"
  >
    <div
      className="inlineclr row"
    >
      <div
        className="col-md-12"
      >
        <div
          className="card-box pt-0"
        >
          <h4
            className="header-title"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <i
                className="icon-arrow-left mr-3"
              />
            </a>
            Billing Details
          </h4>
          <hr
            className="short"
          />
          <div
            className="Billing-container"
          >
            <div
              className="row"
            >
              <div
                className="col-md-12  billing-buttons"
              >
                <button
                  className="btn btn-primary btn btn-default"
                  disabled={false}
                  icon="cloud-upload"
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    className="glyphicon glyphicon-cloud-upload"
                  />
                  Upgrade Plan
                </button>
                <button
                  className="btn btn-primary btn btn-default"
                  disabled={false}
                  icon="usd"
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    className="glyphicon glyphicon-usd"
                  />
                  Payment Method
                </button>
              </div>
              <div
                className="clearfix"
              />
            </div>
            <div
              className="billing-info-list"
            >
              <div
                className="payment-info row"
              >
                <div
                  className="col-md-6"
                >
                  <div
                    className="panel panel-default"
                  >
                    <div
                      className="card-box"
                    >
                      <h4
                        className="header-title mt-0 m-b-20"
                      >
                        Plan details
                      </h4>
                      <div
                        className="panel-body"
                      >
                        <hr />
                        <div
                          className="text-left"
                        >
                          <div
                            className="text-muted font-13"
                          >
                            <strong>
                              Current Plan : 
                            </strong>
                            <span
                              className="m-l-15"
                            >
                              <div
                                className="form-group"
                              >
                                <input
                                  className="form-control mt-4"
                                  disabled={true}
                                  id="campaignname"
                                  type="text"
                                  value=""
                                />
                              </div>
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="col-md-6"
                >
                  <div
                    className="panel panel-default"
                  >
                    <div
                      className="card-box"
                    >
                      <h4
                        className="header-title mt-0 m-b-20"
                      >
                        Payment Info 
                      </h4>
                      <div
                        className="panel-body"
                      >
                        <hr />
                        <div
                          className="text-left"
                        >
                          <p
                            className="text-muted font-13"
                          >
                            <strong>
                              Next Payment Due Date : 
                            </strong>
                            <span
                              className="m-l-15"
                            >
                              -
                            </span>
                          </p>
                          <p
                            className="text-muted font-13"
                          >
                            <strong>
                              Payment Method : 
                            </strong>
                            <span
                              className="m-l-15"
                            >
                              -
                            </span>
                          </p>
                          <p
                            className="text-muted font-13"
                          >
                            <strong>
                              Last Paid : 
                            </strong>
                            <span
                              className="m-l-15"
                            >
                              -
                            </span>
                          </p>
                          <p
                            className="text-muted font-13"
                          >
                            <strong>
                              Billing Cycle : 
                            </strong>
                            <span
                              className="m-l-15"
                            >
                              -
                            </span>
                          </p>
                          <p
                            className="text-muted font-13"
                          >
                            <strong>
                              Visitor Quota : 
                            </strong>
                            <span
                              className="m-l-15"
                            >
                              -
                               Unique Visitors
                            </span>
                          </p>
                          <p
                            className="text-muted font-13"
                          >
                            <strong>
                              Visitor Quota Left : 
                            </strong>
                            <span
                              className="m-l-15"
                            >
                              -
                               Unique Visitors
                            </span>
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="panel panel-default"
              >
                <div
                  className="card-box"
                >
                  <h4
                    className="header-title mt-0 m-b-20"
                  >
                    Invoices 
                  </h4>
                  <div
                    className="col-md-12"
                  >
                    <div
                      className="text-left"
                    >
                      <table
                        className="table-striped table"
                      >
                        <thead>
                          <tr>
                            <th
                              className=" h6"
                            >
                              Billing Date
                            </th>
                            <th
                              className=" h6"
                            >
                              Amount
                            </th>
                            <th
                              className=" h6"
                            >
                              Transaction Id
                            </th>
                            <th
                              className=" h6"
                            >
                              Interval
                            </th>
                            <th
                              className=" h6"
                            >
                              Download Invoice
                            </th>
                          </tr>
                        </thead>
                        <tbody>
                          <tr>
                            nothing
                          </tr>
                        </tbody>
                      </table>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
